### PR TITLE
Add more info about install and uploading

### LINKED ISSUE #9: add info about pip in readme (just simple info), later it can be elengetified

### PR DESCRIPTION
**Pull request by Guseyn**
resolve #9

### CODE REVIEW

**Comment by g4s8** [comment]
@0crat assign me

**Comment by 0crat** [comment]
I'm not managing `artipie/pypi-adapter` GitHub repository, you have to contact me in Slack first, see [§11](http://www.zerocracy.com/policy.html#11)

**Comment by g4s8** [comment]
@0crat in

**Comment by Guseyn** [comment]
@g4s8 @yegor256 could please someone start proceeding this pr?

**Comment by amihaiemil** [comment]
@g4s8 Assign me, I have time now :)

**Comment by Guseyn** [comment]
@amihaiemil I assigned you

**Comment by g4s8** [comment]
@0crat assign @amihaiemil

**Comment by 0crat** [comment]
> @0crat assign @amihaiemil  ([here](https://github.com/artipie/pypi-adapter/issues/5#issuecomment-581870174))

@g4s8 There is an unrecoverable failure on my side. Please, submit it [here](https://github.com/zerocracy/farm/issues):

```
PID: 11@377a78bde909, thread: PQ-CT2E6TK9B
com.zerocracy.claims.ClaimIn[214] java.lang.IllegalArgumentException: Parameter "age" not found in "Order was canceled" among: [comment, job, issue, repo, cause, _parent_sig, flow, login]
```

[0.51.0](https://github.com/zerocracy/farm): CID: [518ce9bd-f4aa-4cd4-a457-40ec9c036f2c](https://www.0crat.com/footprint/CT2E6TK9B/518ce9bd-f4aa-4cd4-a457-40ec9c036f2c), Type: "Order was canceled"

**Comment by 0crat** [comment]
> @0crat assign @amihaiemil  ([here](https://github.com/artipie/pypi-adapter/issues/5#issuecomment-581870174))

@g4s8 This pull request [#5](https://github.com/artipie/pypi-adapter/issues/5) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/4626fb00-1547-4147-a00d-c20d9340089f); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

File: README.md [hunk 1]
```diff
@@ -41,37 +41,59 @@ $ mvn clean install -Pqulice
...
 ```
 
-You can find sample files in the [test resources](https://github.com/artipie/pypi-adapter/tree/master/src/test/resources/simple-pypi-project)
+```
+
+2. For installing run the command
+
+```python
+pip install -i http://localhost:8080/ --upgrade example-pkg-YOUR-USERNAME-HERE
+```
```
Comment (amihaiemil) [inline README.md]: @Guseyn Why did you add the below empty files? ``/example_pkg/README.md`` and ``/example_pkg/__init__.py``? Maybe you can write some dummy texts inside of them so their purpose becomes clearer?

File: README.md [hunk 2]
```diff
@@ -41,37 +41,59 @@ $ mvn clean install -Pqulice
 
 To avoid build errors use Maven 3.2+.
 
-## How to use
+## How it works
 
-For uploading using pip one must create a valid ~/.pypirc file:
+Follow these instructions: [https://packaging.python.org/tutorials/packaging-projects/](https://packaging.python.org/tutorials/packaging-projects/)
+
+1. For uploading using pip one must create a valid ~/.pypirc file:
```
Comment (amihaiemil) [inline README.md]: @Guseyn Can you make this a Header using ``###`` (it will be a Header with a lower case than the parent ``How it works``). 

And can you make it simpler: just ``Uploading using pip`` is enough. The statement ``You must create a valid ~/.pypric file`` can be the first sentence under the header.

File: README.md [hunk 3]
```diff
@@ -41,37 +41,59 @@ $ mvn clean install -Pqulice
...
+And then run the command: `twine upload --repository testpypi dist/*`
+
+Response:
+
 ```
 
-You can find sample files in the [test resources](https://github.com/artipie/pypi-adapter/tree/master/src/test/resources/simple-pypi-project)
+```
+
+2. For installing run the command
```
Comment (amihaiemil) [inline README.md]: @Guseyn This should also be a Header (``##``) and I think it should be: ``Installing a Package using pip``

File: README.md [hunk 4]
```diff
@@ -41,37 +41,59 @@ $ mvn clean install -Pqulice
...
-For uploading using pip one must create a valid ~/.pypirc file:
+Follow these instructions: [https://packaging.python.org/tutorials/packaging-projects/](https://packaging.python.org/tutorials/packaging-projects/)
+
+1. For uploading using pip one must create a valid ~/.pypirc file:
 
 ```
 [distutils]
-index-servers = example
+index-servers=
+    testpypi
```
Comment (amihaiemil) [inline README.md]: @Guseyn The indentation here seems strange. Why not in one line? ``index-servers = testpypi`` ?

File: README.md [hunk 5]
```diff
@@ -41,37 +41,59 @@ $ mvn clean install -Pqulice
...
 
-[example]
-repository = https://example.com/pypi
-username = myname
-password = mypass
+[testpypi]
+repository: http://localhost:8080/
```
Comment (amihaiemil) [inline README.md]: @Guseyn Is the ``repository`` keyword needed for PiP? If it can be anything, I would name it ``url``.

File: README.md [hunk 6]
```diff
@@ -41,37 +41,59 @@ $ mvn clean install -Pqulice
...
+Then run the commands:
 
 ```
-[global]
-extra-index-url = https://myname:mypass@example.com/pypi/simple
+python setup.py sdist
 ```
 
-For using a private server with setuptools unittesting you need to add the following to your setup.py:
+Where `setup.py` looks smth like:
```
Comment (amihaiemil) [inline README.md]: @Guseyn Please don't use Internet abbreviations in READMEs. ``something like this`` instead of ``smth like``. 

Also, can you give a little more context to these examples? For instance, in the ``uploading`` section, I understand you have to run your own repository server, but the ``setup.py`` could use more details: what project are we actually uploading in this example?

**Comment by amihaiemil** [comment]
@Guseyn Also, please link the Github Issue that this PR is solving and provide a decent description of this PR. Something like.

File: README.md (hunk 1 above)
Comment (Guseyn) [reply README.md]: @amihaiemil because it's sample project for future tests, probably I will remove it later. I just need it right now.

File: README.md (hunk 2 above)
Comment (Guseyn) [reply README.md]: I will structure the information in a better way later, it's just a reminder to me what kinda instructions that I need.

File: README.md (hunk 3 above)
Comment (Guseyn) [reply README.md]: I will structure the information in a better way later, it's just a reminder to me what kinda instructions that I need.

Comment (Guseyn) [reply README.md]: @amihaiemil right now it just information, I will make readme document more elegant later.

File: README.md (hunk 4 above)
Comment (Guseyn) [reply README.md]: because it's the format that I found in docs.

File: README.md (hunk 5 above)
Comment (Guseyn) [reply README.md]: @amihaiemil  Yes, it's configuration property, so it must be 'repository'

File: README.md (hunk 6 above)
Comment (Guseyn) [reply README.md]: @amihaiemil  I will fix all stuff later, right now I need such information for further development.

**Comment by Guseyn** [comment]
@amihaiemil plz approve the changes

**Comment by amihaiemil** [comment]
@Guseyn Why "fix them later"? If you wanted to fix stuff later, why is Code Review necessary then? 

Please fix them now. At least the headers stuff and the abbreviations...

**Comment by amihaiemil** [comment]
@Guseyn Again, please don't imagine you are the only developer here. Maybe it's just you at the moment, but this project will have more DEVs probably. So don't do stuff just for yourself like "leaving reminders"...

**Comment by Guseyn** [comment]
@amihaiemil  "please don't imagine you are the only developer here" => I don't imagine. Because I am the only developer here, and I explained that the information that I added is row, and decided format it and structure it in other pr. 
"why is Code Review necessary then?" this question is not for me. I just added info in readme file, but never mind. I close pr.

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Job was finished in 1 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
This pull request [#5](https://github.com/artipie/pypi-adapter/issues/5) is assigned to @fanifieiev[/z](https://www.0crat.com/u/fanifieiev), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/2b2a3a42-81e1-48fb-9b88-a6561a16b018); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Pull request [#5](https://github.com/artipie/pypi-adapter/issues/5) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Comment by 0crat** [comment]
Job [#5](https://github.com/artipie/pypi-adapter/issues/5) is already in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by g4s8** [comment]
@0crat refuse

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
> @0crat out ([here](https://github.com/artipie/pypi-adapter/issues/5#issuecomment-582269489))

@g4s8 Job `gh:artipie/pypi-adapter#5` is not assigned, can't get performer

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/pypi-adapter/issues/5#issuecomment-582269465))

@g4s8 Job `gh:artipie/pypi-adapter#5` is not assigned, can't get performer

**Comment by 0crat** [comment]
This job is not in scope
